geode-core/src/main/java/org/apache/geode/cache/client/internal/ClientMetadataService.java

```diff
@@ -98,11 +96,13 @@ public ClientMetadataService(Cache cache) {
   private final Map<String, Set<ClientPartitionAdvisor>> colocatedPRAdvisors =
       new ConcurrentHashMap<>();
 
-  private PartitionResolver getResolver(Region r, Object key, Object callbackArgument) {
+  @SuppressWarnings("unchecked")
```

Comment:
What's the philosophy vis-à-vis `@SuppressWarnings("unchecked")` in this PR? I see how it suppresses the warnings. On the other hand a deeper revision could potentially eliminate the need for the casts TBD.

Is the idea to identify and suppress all the warnings to get to zero warnings so that we can then avoid introducing more, and to then move forward with separate projects to do the work necessary to actually eliminate the casts (and thereby be able to eliminate the annotations entirely)?

Reply:
In this particular case this method is taking raw values and converting them to the generic typed version requested. Like many other methods that have to do this we need to suppress that unchecked conversion. It is a necessary evil of Java's broken generics system. 

In general this annotation should be reserved for places where it is absolutely required by the language. In some cases it is used where refactoring the callee would introduce a breaking API change or balloon the cleanup too far out of scope. To get any progress we must be ok with the dirty edges along the way to cleaner code.